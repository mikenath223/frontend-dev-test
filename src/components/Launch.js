import styles from '../styles/Launch.module.css';

const Launch = ({ name, image, launchDate, cores }) => {
  return (<li className={styles.launchInfo}>
    <div>
      <img src={image} alt="launch image" />
      <h3>{name}</h3>
    </div>
    <div className={styles.cores}>
      {cores.map(e => <div>
        <h4>{e.core}</h4>
        <p>{e.landing_success?.toString() ?? 'landing success is unknown'}</p>
      </div>)}
    </div>
    <time datetime={launchDate}>{launchDate}</time>
  </li>)
}

export default Launch;