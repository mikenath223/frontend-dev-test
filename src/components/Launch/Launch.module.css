.launchInfo {
  border: 1px solid grey;
  border-radius: 10px;
  list-style: none;
  text-align: center;
}

.cores {
  border: 1px solid red;
  margin: 15px
}

.cores > h4 {
  background-color: grey;
  margin-top: 0
}