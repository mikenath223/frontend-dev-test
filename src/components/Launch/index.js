import PropTypes from 'prop-types';
import styles from './Launch.module.css';

const Launch = ({
  name, image, launchDate, cores,
}) => (
  <li className={styles.launchInfo}>
    <div>
      <img src={image} alt="spacelaunch" />
      <h3>{name}</h3>
    </div>
    <ul className={styles.cores}>
      <h4>Cores Landing Success</h4>
      {cores.map(e => (
        <li key={e.core}>
          <span>
            {e.core}
            :
            {' '}
          </span>
          <span>{e.landing_success?.toString() ?? 'landing success is unknown'}</span>
        </li>
      ))}
    </ul>
    <time dateTime={launchDate}>{launchDate}</time>
  </li>
);

Launch.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  launchDate: PropTypes.string.isRequired,
  cores: PropTypes.arrayOf(PropTypes.shape({
    core: PropTypes.string.isRequired,
    landing_success: PropTypes.bool,
  })).isRequired,
};

export default Launch;
