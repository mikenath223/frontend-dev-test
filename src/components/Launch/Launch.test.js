import { render } from '@testing-library/react';
import Launch from './index';

describe('Launch', () => {
  it('should match snapshot', () => {
    const props = {
      name: 'SatPad',
      image: 'image',
      launchDate: '12/11/2021',
      cores: [],
    };

    // eslint-disable-next-line
    const { container } = render(<Launch {...props} />);

    expect(container).toMatchSnapshot();
  });
});
