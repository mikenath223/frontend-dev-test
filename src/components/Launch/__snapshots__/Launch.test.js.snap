// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch should match snapshot 1`] = `
<div>
  <li
    class="launchInfo"
  >
    <div>
      <img
        alt="spacelaunch"
        src="image"
      />
      <h3>
        SatPad
      </h3>
    </div>
    <ul
      class="cores"
    >
      <h4>
        Cores Landing Success
      </h4>
    </ul>
    <time
      datetime="12/11/2021"
    >
      12/11/2021
    </time>
  </li>
</div>
`;
